src/material/core/theming/_theming.scss

```diff
@@ -14,6 +14,10 @@ $_generate-default-density: true !default;
 $_duplicate-warning: 'Read more about how style duplication can be avoided in a dedicated ' +
   'guide. https://github.com/angular/components/blob/main/guides/duplicate-theming-styles.md';
 
+// Warning that will be printed if the legacy theming API is used.
+$_legacy-theme-warning: 'Themes should be defined by providing a map configuration. ' +
```

Comment:
Should we elaborate more in the warning message? I'm not sure that somebody unfamiliar would know what we mean by "providing a map configuration". They'd also have to scroll pretty far down in the guide to see an example.

Reply:
Updated to: 

Themes should be created from a map containing the keys "color", "typography", and "density". The color value should be a map containing the palette values for "primary", "accent", and "warn". See https://material.angular.io/guide/theming for more information.

Reply:
Sounds good. One last nit is that maybe we should say "Angular Material themes" instead of just "Themes" so it's obvious where the warning is coming from.

Reply:
Done - Looks good, thanks